src/material/slider/slider.ts

```diff
@@ -289,368 +344,901 @@ export class MatSliderVisualThumb implements AfterViewInit, OnDestroy {
-    '[class._mat-animation-noopable]': '_noopAnimations',
-  },
-  exportAs: 'matSlider',
-  changeDetection: ChangeDetectionStrategy.OnPush,
-  encapsulation: ViewEncapsulation.None,
-  inputs: ['color', 'disableRipple'],
-})
-export class MatSlider
-  extends _MatSliderMixinBase
-  implements AfterViewInit, CanDisableRipple, OnDestroy
-{
-  /** The slider thumb(s). */
-  @ViewChildren(MatSliderVisualThumb) _thumbs: QueryList<MatSliderVisualThumb>;
+  _calcTranslateXByValue(): number {
+    if (this._slider._isRtl) {
+      return (1 - this.percentage) * this._slider._cachedTrackWidth + this._slider._inputOffset;
+    }
+    return this.percentage * this._slider._cachedTrackWidth + this._slider._inputOffset;
+  }
 
-  /** The active section of the slider track. */
-  @ViewChild('trackActive') _trackActive: ElementRef<HTMLElement>;
+  _calcTranslateXByPointerEvent(event: PointerEvent): number {
+    return event.pageX - this._slider._cachedLeft;
+  }
 
-  /** The sliders hidden range input(s). */
-  @ContentChildren(MatSliderThumb, {descendants: false})
-  _inputs: QueryList<MatSliderThumb>;
+  _updateHiddenUI(): void {
+    this._updateThumbUIByValue();
+    this._updateWidthInactive();
+  }
 
-  /** Whether the slider is disabled. */
-  @Input()
-  get disabled(): boolean {
-    return this._disabled;
+  // TODO(wagnermaciel): describe the difference between inactive and active width and why we need it.
+  _updateWidthActive(): void {
+    this._paddingStyle = `0 ${this._slider._inputPadding}px`;
+    this._widthStyle = `calc(100% - ${this._slider._inputPadding * 2}px)`;
   }
-  set disabled(v: BooleanInput) {
-    this._setDisabled(coerceBooleanProperty(v));
-    this._updateInputsDisabledState();
+
+  // TODO(wagnermaciel): describe the difference between inactive and active width and why we need it.
+  _updateWidthInactive(): void {
+    this._paddingStyle = '0px';
+    this._widthStyle = '100%';
   }
-  private _disabled: boolean = false;
 
-  /** Whether the slider displays a numeric value label upon pressing the thumb. */
-  @Input()
-  get discrete(): boolean {
-    return this._discrete;
+  _updateThumbUIByValue(options?: {withAnimation: boolean}): void {
+    this.translateX = this._clamp(this._calcTranslateXByValue());
+    this._updateThumbUI(options);
   }
-  set discrete(v: BooleanInput) {
-    this._discrete = coerceBooleanProperty(v);
+
+  _updateThumbUIByPointerEvent(event: PointerEvent, options?: {withAnimation: boolean}): void {
+    this.translateX = this._clamp(this._calcTranslateXByPointerEvent(event));
+    this._updateThumbUI(options);
   }
-  private _discrete: boolean = false;
 
-  /** Whether the slider displays tick marks along the slider track. */
-  @Input()
-  get showTickMarks(): boolean {
-    return this._showTickMarks;
+  _updateThumbUI(options?: {withAnimation: boolean}) {
+    this._slider._transition =
+      options?.withAnimation && !this._slider._noopAnimations ? 'transform 80ms' : 'transform 0ms';
```

Comment:
I wasn't able to do it using CSS classes. I think because mdc has some very specific classes to set the transition to 80ms. I instead directly updated the styles in JS

Reply:
Never mind, I got it working with classes